### ISSUE TITLE
OverridableMethodInConstructor warns on code which actually doesn't invoke overridable method

### ISSUE THREAD

**Issue by baratali** [sevntu-checkstyle/sevntu.checkstyle#563]
Test java file:
``$ cat Test.java``
```java
import java.awt.event.ActionListener;

class Test {
	ActionListener listener;

	public Test() {
		listener = event -> process();
	}

	// can be overriden
	public void process() {
		System.out.println("Processing...");
	}

	public static void main(String[] args) {
		new Test();
	}
}
```
Config file:
``$ cat checkstyle-config.xml``
```xml
<?xml version="1.0" ?>

<!DOCTYPE module PUBLIC
        "-//Puppy Crawl//DTD Check Configuration 1.2//EN"
        "http://www.puppycrawl.com/dtds/configuration_1_2.dtd">

<module name="Checker">
    <module name="TreeWalker">
        <module name="OverridableMethodInConstructor" />
    </module>
</module>
```
Checkstyle's output:
```
$ java -cp checkstyle-7.6-all.jar:sevntu-checks-1.23.1.jar com.puppycrawl.tools.checkstyle.Main -c checkstyle-config.xml Test.java
Starting audit...
[ERROR] /home/bizmailov/test/Test.java:7:44: Overridable method 'process' is called in constructor body. [OverridableMethodInConstructor]
Audit done.
Checkstyle ends with 1 errors.
```

However, method ``process()`` is mentioned in lambda (or anonymous class) and never invoked.
```
$ javac Test.java 
$ java Test
$
```
